src/Calendar.elm

```diff
@@ -133,15 +134,16 @@ view state =
             ]
 
 
-dayRangeOfWeek date =
+weekRangeFromDate : Date -> List Date
+weekRangeFromDate date =
     let
-        firstOfWeek =
-            Date.Extra.floor Date.Extra.Week date
+        startOfWeek =
+            Date.Extra.floor Date.Extra.Sunday date
     in
         Date.Extra.range Date.Extra.Day
             1
-            (Date.Extra.floor Date.Extra.Sunday firstOfWeek)
-            (Date.Extra.ceiling Date.Extra.Sunday firstOfWeek)
```

Comment:
In Date.Extra, `floor` and `ceiling` work like they do on numbers, so if the given date is already at a rounded interval, then both functions will return the same date (e.g. `(floor 7.0, ceiling 7.0) == (7, 7)`). Instead, you want to `floor` for the start date, and then `add` to that for the end date.